### PR TITLE
[#93] Enhance commit messages (part 3)

### PR DESCRIPTION
**Pull request by pmeinhardt**
- [x] add checkbox for applying recommended format on options page
- [x] conditionally apply formatting to commit message preview (when enabled)
- [x] conditionally apply formatting to commit message (when enabled)
- [x] add test for form with auto-format toggling
- [x] reference tpope’s recommendation post in options page
- [x] reference blog post in readme
- [x] fix checkbox style in Firefox
- [x] support Safari settings

https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

preview

### CODE REVIEW

**Comment by pmeinhardt** [comment]
🤔 Chrome seems to have issues with rendering the checkbox tick on options pages.

screenshot

**Comment by pmeinhardt** [comment]
Fixed the checkbox issue by updating the extension security policy (see [`MDN: CSP img-src`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src)).

ticked checkbox

Chrome refused to load the checkbox tick mark because the browser stylesheet implements it through a pseudo-element with a `data:` URI. Such URIs were previously not allowed by our content security policy defined in the extension manifest.

**Comment by pmeinhardt** [comment]
Here's a slightly improved version. Now Firefox renders it slightly differently, but I think this is the right direction.

chrome

firefox

**Comment by pmeinhardt** [comment]
FYI: I've added a preview in the issue description above

File: src/web-extension/options/components/form.test.jsx [hunk 1]
```diff
@@ -67,6 +77,8 @@ describe('form', () => {
     });
   });
 
+  it.todo('conditionally pretty-prints the commit message');
```
Comment (pmeinhardt) [inline src/web-extension/options/components/form.test.jsx]: TODO

**Comment by klappradla** [comment]
Ping reviewers (@andreasknoepfle / @lwassermann / @nickgnd) anyone with additional opinions on this here? Otherwise I'd probably go ahead and yolo-merge... 🤔

**Pull request merged**